docs/guides/social-signin/github.mdx

```diff
@@ -0,0 +1,89 @@
+
+Click to enable GitHub, which will take you to a form that requires you to set a
+few values. For now, just copy the "Redirect URI".
+
+Now, if you haven't already, you must create a
+[GitHub OAuth2 App](https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/).
+
+When you get to the app creation screen, you are free to choose the values -
+except for "Authorization callback URL" which is the URL you just copied.
+
+![Create Personal Access Token](./screenshots/github-redirect-uri.png)
+
+Click "Register application", after which:
+
+- You can copy the Client ID, and paste it into the form in Ory Cloud Console
+- Generate a new Client Secret (which you should keep secret), and paste that
+  into the form as well
+
+### Scopes
+
+These are the OAuth scopes to be requested. This is handy when you plan to
+interact with the third party's APIs on behalf of the user, or when you need
+access to additional user data (such as email addresses), which Ory then exposes
+as claims in the data mapping in the next step. Leave blank for default scope.
+
+All available scopes for GitHub can be found
+[in their docs](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps).
+
+### Data Mapping
+
+You may leave this blank, but user experience may be worse for it, as a user
+will have to supply their email address after completing the OAuth flow.
+
+As explained in
+[OpenID Connect and OAuth2 Credentials](../../kratos/concepts/credentials/openid-connect-oidc-oauth2.mdx),
+you can add a Jsonnet code snippet for the provider, so Ory can map the data
+returned by it to the traits specified in your identity schema.
+
+:::note
+
+GitHub doesn't implement OpenID Connect. Therefore, Ory Kratos makes a request
+to
+[GitHub's User API](https://developer.github.com/v3/users/#get-the-authenticated-user)
+and adds that data to `std.extVar('claims')`. Check out what data is available
+at
+[GitHub's Scope Docs](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/).
+However, not all GitHub fields are supported. Check the list of supported fields
+[in Kratos' source code](https://github.com/ory/kratos/blob/v0.2.1-alpha.1/selfservice/strategy/oidc/provider_github.go#L72-L98).
+
+:::
+
+The following Jsonnet takes `email_primary` and maps it to `traits.email`:
+
+```github.jsonnet"
+local claims = {
+  email_verified: false
+} + std.extVar('claims');
+
+{
+  identity: {
+    traits: {
+      // Allowing unverified email addresses enables account
+      // enumeration attacks, especially if the value is used for
+      // e.g. verification or as a password login identifier.
+      //
+      // Therefore we only return the email if it (a) exists and (b) is marked verified
+      // by GitHub.
+      [if "email" in claims && claims.email_verified then "email" else null]: claims.email,
+    },
+  },
+}
+```
+
+### Wrapping up
+
+After you submit, GitHub will be good to go!
```

Comment:
```suggestion
---
id: github
title: GitHub
---

Follow these steps to add GitHub as a social sign-in provider for your Ory Cloud project. 

:::warning

Currently, you can't remove the social sign-in providers you added.

:::

1. Sign in to the Ory Cloud Console and select **Social Sign-in**.
2. Click the button next to the GitHub logo to start the configuration. 
3. Copy the Redirect URI. 
3. Create an OAuth2 app in GitHub. Open a new browser tab, go to [Developer settings → OAuth Apps](https://github.com/settings/developers) and click **New OAuth App**.
4. Paste the Redirect URI copied from Ory into the **Authorization Callback URL** field in GitHub and fill in other required app details.
5. Click the **Register application** button. 
6. Copy the Client ID of the registered application and paste it into the corresponding field in the Ory Console.
7. In GitHub, click **Generate a new client secret**, copy the generated string, and paste it into the corresponding field in the Ory Console.
8. Click **Save Configuration** to enable the social sign-in provider. 

:::note

The **Scopes** and **Data mapping** fields are optional - you don't have to configure them to start using the social sign-in provider.

:::

## Additional configuration

When adding a social sign-in provider, you can customize the integration by defining the OAuth scopes Ory requests from the provider
and by setting up custom data mappings.

### Scopes

The Scopes section allows you to define the OAuth scopes Ory requests from the sign-in provider. 
Defining scopes allows you to interact with the provider's APIs on behalf of the user or to access additional
user data, which is exposed as claims for data mapping. 

:::tip

Leave this field blank to use the default scope.

:::

To learn more about the scopes available in GitHub, read the [related documentation](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps).

### Data Mapping

The Data Mapping section allows you to map the data returned by the sign-in provider as claims to
traits defined in the identity schema.

To define the mapping, create a Jsonnet code snippet. Read the [OpenID Connect and OAuth2 Credentials](../../kratos/concepts/credentials/openid-connect-oidc-oauth2.mdx) documentation for more details.

:::warning

GitHub doesn't implement OpenID Connect. Because of this limitation, Ory Kratos makes a request
to
[GitHub's User API](https://developer.github.com/v3/users/#get-the-authenticated-user)
and adds the user data to `std.extVar('claims')`. Learn what data is available
by reading the
[GitHub Scopes documentation](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/).
However, not all GitHub scopes are supported. Check the list of supported fields
[the source code](https://github.com/ory/kratos/blob/v0.2.1-alpha.1/selfservice/strategy/oidc/provider_github.go#L72-L98).

:::

In this sample Jsonnet snippet, `email_primary` is mapped to the identity schema's `traits.email`:

```github.jsonnet"
local claims = {
  email_verified: false
} + std.extVar('claims');

{
  identity: {
    traits: {
      // Allowing unverified email addresses enables account
      // enumeration attacks, especially if the value is used for
      // e.g. verification or as a password login identifier.
      //
      // Therefore we only return the email if it (a) exists and (b) is marked verified
      // by GitHub.
      [if "email" in claims && claims.email_verified then "email" else null]: claims.email,
    },
  },
}
```

```